compiler/qsc_parse/src/lex/cooked.rs

```diff
@@ -51,6 +51,10 @@ pub(crate) enum Error {
     #[error("unrecognized character `{0}`")]
     #[diagnostic(code("Qsc.Lex.UnknownChar"))]
     Unknown(char, #[label] Span),
+
+    #[error("unfinished generic")]
+    #[diagnostic(code("Qsc.Lex.UnfinishedGeneric"))]
+    UnfinishedGeneric(#[label] Span),
```

Comment:
This PR isn’t consistent about whether the token is referred to as an “apostrophe identifier” or a “generic.” I would recommend picking one. The lexer prefers low-level descriptions like apostrophe, while the parser uses high-level descriptions like generic parameter.